### PR TITLE
include item birthtime in archive (where available)

### LINKED ISSUE #3272: macOS creation dates are not preserved
## Description

macOS creation dates are not preserved.

## Environment

* macOS 10.12.6
* Borg git @ d1d55904

## Steps to replicate

1. Prepare a file that has a known creation date and last modified date
```
mkdir test; cd test
mkdir src; cd src
touch dated-file
SetFile -d 12/31/1999 dated-file
SetFile -m 10/10/2000 dated-file
cd ..
```

2. Back up the file:
```
sudo borg init -e repokey sam@keep:/storage/backupdisk/borg/date-test
sudo borg create sam@keep:/storage/backupdisk/borg/date-test::test1 src
```

3. Restore the file:
```
mkdir dst; cd dst
sudo borg extract --strip-components=1 sam@keep:/storage/backupdisk/borg/date-test::test1 src
cd -
```

## Expected result

* Resulting file has a creation date of 12/31/1999.
    * Verify this with `GetFileInfo -d dst/dated-file`.

## Actual result

* Resulting file has a creation date of 10/10/2000.

## Notes

* It looks like borg either doesn't set the creation date at all or sets it to the modification date.
* If this was resolved, borg would pass all my tests for preserving macOS metadata (a feat so far only achieved by CrashPlan).

### PR DESCRIPTION
**Pull request by smammy**
This adds birth times (macOS creation date) to the archive, to resolve #3272. I'm new to this codebase so I expect there will be things I messed up. I took a wild guess that adding keys to `Item` was OK and that a new (post-1.1) `int` field didn't need a `bigint` wrapper.

### CODE REVIEW

**Comment by smammy** [comment]
If someone could help me understand the Travis failures I'd appreciate it.

File: src/borg/constants.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 ITEM_KEYS = frozenset(['path', 'source', 'rdev', 'chunks', 'chunks_healthy', 'hardlink_master',
                        'mode', 'user', 'group', 'uid', 'gid', 'mtime', 'atime', 'ctime', 'size',
                        'xattrs', 'bsdflags', 'acl_nfs4', 'acl_access', 'acl_default', 'acl_extended',
-                       'part'])
+                       'part', 'birthtime'])
```
Comment (ThomasWaldmann) [inline src/borg/constants.py]: this is sorted by topic, so please move it to after ctime.

File: src/borg/item.pyx [hunk 2]
```diff
@@ -169,6 +169,9 @@ class Item(PropDict):
     ctime = PropDict._make_property('ctime', int, 'bigint', encode=int_to_bigint, decode=bigint_to_int)
     mtime = PropDict._make_property('mtime', int, 'bigint', encode=int_to_bigint, decode=bigint_to_int)
 
+    # birthtime is new since 1.1, so it doesn't need a bigint wrapper (i think)
+    birthtime = PropDict._make_property('birthtime', int)
```
Comment (ThomasWaldmann) [inline src/borg/item.pyx]: guess we would rather do it consistently with the others.

File: src/borg/archive.py [hunk 3]
```diff
@@ -925,6 +938,9 @@ def stat_simple_attrs(self, st):
             attrs['atime'] = safe_ns(st.st_atime_ns)
         if not self.noctime:
             attrs['ctime'] = safe_ns(st.st_ctime_ns)
+        if not self.nobirthtime and hasattr(st, 'st_birthtime'):
+            # sadly, there's no stat_result.st_birthtime_ns
+            attrs['birthtime'] = int(st.st_birthtime * 10**9)
```
Comment (ThomasWaldmann) [inline src/borg/archive.py]: safe_ns() here also to be consistent?

**Comment by smammy** [comment]
OK, I made all the changes you suggested and added tests. I'm not sure how to interpret the code coverage output though, so some guidance would be appreciated.

**Comment by smammy** [comment]
OK, I made all the changes you suggested and added tests. I'm not sure how to interpret the code coverage output though, so some guidance would be appreciated.

**Comment by ThomasWaldmann** [comment]
looks good, thanks!

**Comment by ThomasWaldmann** [comment]
can you backport to 1.1-maint (see squashed commit), test it and do a PR?

**Pull request merged**